### PR TITLE
[WIP] - #2501 - Retries RPC call when node fails

### PR DESCRIPTION
**Pull request by Moejoe90**

### CODE REVIEW

File: app/lib/web3/Web3Batch/Web3Batch.ts [hunk 1]
```diff
@@ -26,9 +29,19 @@ class Web3AutoExecuteBatch {
...
-
-    this.web3Batch.execute();
-    this.web3Batch = undefined;
+    for (let retries = 0; retries < NUMBER_OF_ALLOWED_RETIRES; retries++) {
+      try {
+        this.web3Batch.execute();
+        this.web3Batch = undefined;
+        return;
+      } catch (e) {
+        delay(FAILED_REQUEST_WAIT_TIME);
```
Comment (desfero) [inline app/lib/web3/Web3Batch/Web3Batch.ts]: can we add a log here? Can be useful in some cases for debugging and monitoring how often retries are called
```
this.logger.info(`${retries} retry call to node with ${this.web3Batch.requests.length}` to batch);
```
and then move log from line 31 below line 35 so structure is consistent

File: app/lib/web3/Web3Batch/Web3Batch.ts [hunk 2]
```diff
@@ -26,9 +29,19 @@ class Web3AutoExecuteBatch {
 
   private execute = () => {
     this.logger.info(`Number of web3 node rpc request batched: ${this.web3Batch.requests.length}`);
-
-    this.web3Batch.execute();
-    this.web3Batch = undefined;
+    for (let retries = 0; retries < NUMBER_OF_ALLOWED_RETIRES; retries++) {
```
Comment (desfero) [inline app/lib/web3/Web3Batch/Web3Batch.ts]: can we add unit tests fo this?

File: app/lib/web3/Web3Batch/Web3Batch.ts [hunk 3]
```diff
@@ -26,9 +29,19 @@ class Web3AutoExecuteBatch {
...
+        this.web3Batch.execute();
+        this.web3Batch = undefined;
+        return;
+      } catch (e) {
+        delay(FAILED_REQUEST_WAIT_TIME);
+        continue;
+      }
+    }
+    throw new NodeNotRespondingError(
+      `Failed to Connect to RPC Node after ${NUMBER_OF_ALLOWED_RETIRES} retries`,
```
Comment (desfero) [inline app/lib/web3/Web3Batch/Web3Batch.ts]: hm, can we also log the original error in some way? Like move this to `catch` and add original error message?

**Comment by Moejoe90** [comment]
This will get closed until further investigation

**Pull request closed without merging**